src/host/proxy/ITerminalHandoff.idl

```diff
@@ -4,15 +4,63 @@
 import "oaidl.idl";
 import "ocidl.idl";
 
+
+typedef struct _TERMINAL_STARTUP_INFO
+{
+    // In STARTUPINFO
+    LPCWSTR pszTitle;
+
+    // Also wanted
+    LPCWSTR pszIconPath;
+    LONG iconIndex;
+
+    // The rest of STARTUPINFO
+    DWORD dwX;
+    DWORD dwY;
+    DWORD dwXSize;
+    DWORD dwYSize;
+    DWORD dwXCountChars;
+    DWORD dwYCountChars;
+    DWORD dwFillAttribute;
+    DWORD dwFlags;
+    WORD wShowWindow;
+
+    // Something else we may be interested in?
+    LPCWSTR fontFace;

```

Comment:
Interesting. Does this come in during startup? I was thinking that Terminal could just crack the registry or lnk file itself instead of being "told" this.

Reply:
In general, this feels like `YAGNI`. We're adding something speculatively that we might one day want, and every time we do that we accumulate cruft... that stuff almost always goes unused. Prefer adding it when we need it, EVEN IF it means we need to rev the interface in the future.

Reply:
Consider: can we treat the struct like all those Windows structs that has a `cbSize` member? We can use the `cbSize` to determine what version it is, without revving the `ITerminalHandoff2` interface.

Reply:
> can we treat the struct like all those Windows structs that has a `cbSize` member?

I honestly don't know - probably not, right? For those, the recipient of the API call is in proc, can see how big the struct really is, cast and pull out what it needs to tell the OS, yea? In this case, COM is gonna send literally this buffer of stuff over. I don't _think_ we can just pass a blob saying "no wait I'm actually bigger" and have the process on the other side be like "sure I trust you lemme just read some random memory"

Reply:
I'll just yank this

Reply:
> can see how big the struct really is

Eh, not really. At the end of the day, two inproc components can trivially disagree about the size of something -- cf. all of the C language project vulnerabilities ;)

I do suspect it would be bad for RPC, since the RPC proxy is always the "newest version" and would almost certainly try to memcpy() the "newest version size". So, you're right -- but for a different reason. :)